libs/cardano-ledger-test/src/Test/Cardano/Ledger/Examples/Consensus.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Test.Cardano.Ledger.Examples.Consensus (genericConsensusTest) where
+
+import Cardano.Ledger.Crypto (StandardCrypto)
+import Cardano.Ledger.Era (EraCrypto)
+import Cardano.Ledger.Pretty (PDoc, ppString, text)
+import Prettyprinter (vsep)
+import Test.Cardano.Ledger.Generic.Consensus (ledgerExamples, oldLedgerExamples)
+import Test.Cardano.Ledger.Generic.Proof (Evidence (..), Proof (..), Reflect (..))
+import Test.Cardano.Ledger.Generic.Same (Same (..))
+import Test.Tasty
+import Test.Tasty.HUnit (Assertion, assertBool, testCase)
+
+-- ===============================================================
+
+oneTest :: (Reflect era, EraCrypto era ~ StandardCrypto) => Proof era -> Assertion
+oneTest proof = assertBool (displayResults results) (null results)
+  where
+    old = oldLedgerExamples proof
+    new = ledgerExamples proof
+    equal =
+      ( case proof of
+          Shelley _ -> old == new
+          Allegra _ -> old == new
+          Mary _ -> old == new
+          Alonzo _ -> old == new
+          Babbage _ -> old == new
+          Conway _ -> old == new
+      )
```

Comment:
Unfortunately, this does not work to use specialize because old,new :: ShelleyLedgerExamples era
and ShelleyLedgerExamples  is a type family that cannot be partially applied. 
But we can remove the parenthesis.